sortableList/imports/ui/components/sortableList/sortableList.js

```diff
@@ -26,7 +26,12 @@ Template.sortableList.onCreated(function () {
     }
 
     this.reorderList = (index, newIndex) => {
-        // complete this method
+        const listItems = this.state.get('listItems');
+        const temp = listItems[index];
+        const dir = index < newIndex ? 1 : -1;
+        for(let i = index; i != newIndex; i += dir) listItems[i] = listItems[i + dir];
```

Comment:
here it is appropriate to use splice to manipulate the list
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

using splice you can simply remove temp from the list and insert it at a different position instead of reassigning elements to new indices manually.

In general, for loops with counters are to be avoided. They can be hard to reason about and overly verbose. 

Prefer the use of functions like map, filter, each, etc. to apply a function to each element in a list or collection.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map
https://lodash.com/docs/4.16.6#map

another useful piece of syntax: given a list l, a for loop to print its elements can be written: for (let i in l) console.log(i)